### PR TITLE
#7117: Updated migration guidelines.

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
I updated the migration guidelines for projects. Didn't experienced a migration yet, only reorganized the reported breaking changes accordingly to most recent changes. 

Please let me know from your experience of migrating project if I have to add something.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Documentation

## Issue

**What is the current behavior?**

#7117 

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
geostore version update needs to be added ?

**Comment by offtherailz** [comment]
mmm... I didn't think about it . 
Maybe "check changes in `pom.xml`" includes it. Did anybody migrated to master yet (e.g. c040 or more)?

**Comment by MV88** [comment]
> mmm... I didn't think about it .
> Maybe "check changes in `pom.xml`" includes it. Did anybody migrated to master yet (e.g. c040 or more)?

not yet, yes you are right

File: docs/developer-guide/mapstore-migration-guide.md [hunk 1]
```diff
@@ -73,17 +78,20 @@ We suggest you to move them as well from root to configs folder
...
 - `simple.json`
 
-Back-end has been reorganized
+#### Back-end has been reorganized
+
 In particular:
 
 - all the java code has been moved from `web/src/` to the `java/` and `product/` directories (and `release`, already existing).
 - `mapstore-backend` has been renamed into `mapstore-services`.
+- `geostore` version used is `1.7.0` on release `2021.02.00` (`1.8-SNAPSHOT` on `master` branch)
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:87]:
```suggestion
- `geostore` version used is `1.7.0` on release `2021.02.00`
```

**Comment by offtherailz** [comment]
If you approve @MV88 we can merge

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@offtherailz can you backport  to 2021.02.xx ?

**Pull request merged**